blueprint/db/src/lib.rs

```diff
@@ -41,13 +41,16 @@ pub async fn transaction(
 pub enum Error {
     /// General database error, e.g. communicating with the database failed
     #[error("database query failed")]
-    DbError(anyhow::Error),
-    /// No record was found where one was expected, e.g. when loading a record by ID
-    #[error("no record found where one was expected")]
+    DbError(#[from] sqlx::Error),
+    /// No record was found, e.g. when loading a record by ID. This variant is different from
+    /// `Error::DbError(sqlx::Error::RowNotFound)`, in that the latter indicates a bug, and
+    /// `Error::NoRecordFound` does not. It merely originates from [sqlx::Executor::fetch_optional]
+    /// returning `None`.
+    #[error("no record found")]
```

Comment:
I think it should be the other way around really – the idea here was that if no record being found being kind of expected, we return `None` but this error indicates an actual error (see original error message above)

Reply:
That is interesting, as in the old version of the diff, 'Option::None' resulting from 'fetch_optional' is mapped to 'Error::NoRecordFound', which would indicate the semantics are as described

Reply:
nvm, I think you're right